hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -216,7 +216,15 @@ public ClientConnection getConnection(Address target) {
         if (target == null) {
             return null;
         }
-        return connections.get(target);
+        ClientConnection connection = connections.get(target);
+        if (connection != null && !connection.isAlive()) {
+            if (logger.isFinestEnabled()) {
+                logger.finest("Get a connection " + connection + " which is not alive. Destroying it.");
+            }
+            destroyConnection(connection, "Not Alive", null);
```

Comment:
in general, i think that we still need to find the root causes why those connections can stay dangling like that. The solution will eliminate after affects of such a case and may hide the actual problem why they happen at the first place.